define issue templates for each Issue Type

turadg opened the pull request:
incidental

## Description

[GitHub has been improving their Issues functionality](https://github.blog/changelog/2025-01-13-evolving-github-issues-public-preview/). They now have [Issue Types](https://github.com/orgs/community/discussions/148715).

This adopts them in our templates. Our `work_item` template conflated Features and Tasks so this splits them into two templates. The existing structure seemed best suited for "feature" so this adds a new simpler structure for "task".

### Security Considerations
n/a

### Scaling Considerations
n/a


### Documentation Considerations
Somewhat self-documenting. Once approved by eng leads we'll need to communicate with the team before enacting (by merge).

### Testing Considerations
Would be nice if we could see it in action but not possible until merge. Easy to tweak though.

### Upgrade Considerations
n/a

(I wonder if we should have another PR template for stuff like this that obviously has no upgrade impact.)

mhofman commented on .github/ISSUE_TEMPLATE/task.md:
I think we need to give some guidelines as to what constitutes a task. The worry is that we'd end up with standalone tasks (not sub-tasks) that end up having significant impact on chain behavior, or non-trivial upgrade considerations.

turadg commented:
In draft until we can [see a history of changes](https://github.com/orgs/community/discussions/148715#discussioncomment-11948692)

mhofman commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -2,7 +2,8 @@
 name: Bug report
 about: Create a report to help us improve
 title: ''
-labels: 'bug'
```
Should we keep both the label and the type for now until we transition all old issues (including ones not tracked in projects)?

mhofman reviewed (approved):
LGTM but would love to have others opine.

turadg commented:
I've updated all the [open bug issues](https://github.com/Agoric/agoric-sdk/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug%20no%3Atype) to use the Issue Type: Bug .

So that we don't use the "bug" label anymore I've renamed it to [bug~deprecated](https://github.com/Agoric/agoric-sdk/labels/bug~deprecated). I didn't remove it so that we don't have to convert the hundreds of old issues over. I though this would help with discovery too because its description says "use the new Bug issue type instead". 

The [enhancement label](https://github.com/Agoric/agoric-sdk/issues?q=is%3Aissue%20is%3Aopen%20no%3Atype%20label%3Aenhancement) isn't so automatable because some are Feature and some are Task. So I've left that alone.
